presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilter.java

```diff
@@ -839,6 +850,137 @@ public String toString()
         }
     }
 
+    public static class BytesValues
+            extends AbstractTupleDomainFilter
+    {
+        private final byte[][] values;
+        private final byte[][] hashTable;
+        private final int hashTableSizeMask;
+        long[] bloom;
+        int bloomSize;
+
+        public BytesValues(byte[][] values, boolean nullAllowed)
+        {
+            super(nullAllowed);
+
+            requireNonNull(values, "values is null");
+            checkArgument(values.length > 1, "values must contain at least 2 entries");
+
+            this.values = values;
+            // Linear hash table size is the next power of two greater
+            // than or equal to number of values* 5.
+            int hashTableSize = Integer.highestOneBit(values.length * 5);
+            hashTableSizeMask = hashTableSize - 1;
+            hashTable = new byte[hashTableSize][];
+            // 8 bits of Bloom filter per slot in hash table. The
+            // bloomSize is a count of longs, hence / 8.
+            bloomSize = Math.max(1, hashTableSize / 8);
+            bloom = new long[bloomSize];
+            for (byte[] value : values) {
+                long hashCode = hash(value, 0, value.length);
+                addBloom(hashCode);
```

Comment:
I'd inline this method. It is used only once and it is just one line of implementation.